src/python/pants/backend/project_info/peek.py

```diff
@@ -65,6 +78,16 @@ class PeekSubsystem(Outputting, GoalSubsystem):
         ),
     )
 
+    include_goals = BoolOption(
```

Comment:
Any reason not to always include goals? Are they expensive to compute?

Reply:
Good question - as it calls out to another rule which performs some work, I'm hesitant to force it on anyone using this API. 

The `peek` API feels like one that can/will/does balloon - but just the sheer JSON that someone downstream has to manage if we keep adding new fields to it. 

Probably something we should decide on (Monday meeting topic?), since adding fields in is safe - but removing them is a breaking change.

Reply:
I just don't want to add extra toggles when `peek` already contains so much data, and a typical user of it will only want a subset anyway. I think the argument for toggles (and this holds for the two we have) is that computing the extra data slows things down.

Reply:
With the current structure, adding new fields is not safe, as they're in the same namespace as field values.. I have a plan to re-organize the peek data structure to have the target fields values nested one level deeper, to allow these kinds of changes in a safe way.
https://github.com/pantsbuild/pants/issues/18546

Reply:
@kaos This confuses me - what makes these new goals unsafe? I don't really understand the concern here with outputted JSON

Reply:
@benjyw  To follow up on this, running on 6300 targets, I saw around a 1-2% increase in time with `--include-goals` which is also well within the run-to-run variance on my machine, so I think we can softly conclude this is a mild perf hit at best right now.